### PR TITLE
Address reading non-binary static files in themes

### LINKED ISSUE #5676: Theme assets do not respect static files within the site
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [x] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/

---

- [x] I am on (or have tested on) ***macOS*** 10+

---

- [x] It was another bug.

## My Reproduction Steps

1. `jekyll new theme-test`
2. `cd theme-test`
3. Replace `gem "minima"` with `gem "minimal-mistakes-jekyll"` in `Gemfile`
4. Run `bundle update`
5. Replace `theme: minima` with `theme: minimal-misakes-jekyll` in `_config.yml`
6. Remove `about.md` to suppress missing includes warnings
7. Add a test file named `/assets/js/main.min.js` and a comment `/* testing if this overrides theme gem's file */`
8. Run `bundle exec jekyll build`
9. Examine contents of `_site/assets/js/main.min.js` to see if test comment is there.

## The Output I Wanted

Local `/assets` files should override those found in the bundled gem theme, but they are not. Instead the theme gem's versions are copied to the site regardless if there are local versions `_site`.

The only `/assets` file I've had success overriding is `/assets/css/main.scss`. This file has YAML Front Matter which I thought may be a reason for it working, but adding `--- ---` to any other files has no affect.

### PR DESCRIPTION
**Pull request by ashmaroli**
This is related to #5676. I may have found a solution, but not sure if the approach is correct.
I'm pushing commits separately to trigger unique ci builds for them.

/cc @parkr 

--

Fixes #5676

### CODE REVIEW

**Comment by ashmaroli** [comment]
Can this be released with `v3.4.1` too?

**Comment by parkr** [comment]
> Can this be released with v3.4.1 too?

No, I don't suspect we'll do anything other than the PR I have opened. `master` is pointed to a v3.5 and I'd like to just keep going with that. This feels like a big enough change to be in a minor bump.

**Comment by ashmaroli** [comment]
label: `themes`, `fix`

**Comment by ashmaroli** [comment]
fixes #5676

**Comment by ashmaroli** [comment]
> but that'd be a pain

@zyv Or you could just use this *PR's branch on my fork of Jekyll*, till this is merged and released.

File: lib/jekyll/readers/theme_assets_reader.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def read_theme_asset(path)
           Jekyll::Page.new(site, base, dir, name)
       else
         append_unless_exists site.static_files,
-          Jekyll::StaticFile.new(site, base, dir, name)
+          Jekyll::StaticFile.new(site, base, "/#{dir}", name)
```
Comment (ashmaroli) [inline lib/jekyll/readers/theme_assets_reader.rb]: > ?

@parkr 
A `Jekyll::StaticFile` path in `source_dir` : `/assets/img/logo.png`

A `Jekyll::StaticFile` path in `theme_dir` **without this PR** : `assets/images/hero.jpg`
A `Jekyll::StaticFile` path in `theme_dir` **with this PR** : `/assets/images/hero.jpg`

The above minor discrepancy is what causes #5676

**Comment by ashmaroli** [comment]
`StaticFile` path from the `source_dir` has a trailing slash while the path to a `StaticFile` from a theme-gem doesn't. This is what caused the mentioned issue.

**Comment by ashmaroli** [comment]
:bellhop_bell: :smile_cat:

**Comment by codedbypm** [comment]
+1 would love to see this in the next jekyll release.

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Comment by ashmaroli** [comment]
> I don't love the interpolation of "/#{dir}" because it smells like a potential security issue to me

@parkr Do you want me to pass the variable through an `ensure_leading_slash` method?

**Pull request merged**